src/test/regress/bin/normalize.sed

```diff
@@ -219,6 +219,20 @@ s/^(ERROR:  child table is missing constraint "\w+)_([0-9])+"/\1_xxxxxx"/g
     }
 }
 
+# normalize for random waits for CREATE INDEX CONCURRENTLY isolation tests.
+# <waiting ...> outputs can be in separate lines, or on the same line, and hence
+# we have a slightly more complex pattern.
+# All the flaky tests use a index name that starts with `flaky` so we limit the
+# normalization using that pattern.
+/CREATE INDEX CONCURRENTLY flaky/ {
```

Comment:
do we still want to add normalization rule? Can we try the suggestion @JelteF had on the PR?

Reply:
I tried to do that, but I am having a hard time incorporating this change for checking on the workers if any logical replication job got blocked.

Does `citus_lock_waits` actually contain the information on blocked logical replication jobs as well?

Reply:
I'm fine with merging the current normalization approach if that's an easier way to fix it. Could you add a link to the issue that Onder opened about to the comment though. Then at least if we ever fix it we can remove this normalization rule.

Reply:
I applied the following diff to current master branch, and it seemed to solve the flakiness.

```diff
diff --git a/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql b/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql
index 64b89ec0e..c652a345a 100644
--- a/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql
+++ b/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql
@@ -24,8 +24,7 @@ RETURNS boolean AS $$
     END IF;

     RETURN EXISTS (
-      SELECT 1 FROM citus_internal_global_blocked_processes()
-        WHERE waiting_global_pid = mBlockedGlobalPid
+      SELECT 1 FROM citus_lock_waits WHERE waiting_gpid = mBlockedGlobalPid AND waiting_gpid != blocking_gpid
     ) OR EXISTS (
       -- Check on the workers if any logical replication job spawned by the
       -- current PID is blocked, by checking it's application name
```

`isolation_select_vs_all_on_mx` used to fail around 2 times out of 100 runs, now I did not see any failure in 200 runs.

However I noticed that isolation tests started to take a considerably more time to finish. The time to finish 100 runs increased from 209 seconds to 335 seconds.

I agree that this is an improvement for `citus_isolation_test_session_is_blocked`. However I also think that it is not worth increasing test times this much. What do you think @JelteF @onderkalaci ?

Reply:
I don't think we should use `citus_lock_waits`, it should be fine to keep `citus_internal_global_blocked_processes WHERE blocking_global_pid != waiting_global_pid` ?

Reply:
I tried and tested another change after your comment, and it did not work. 1 test failed out of 200 runs.

```diff
diff --git a/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql b/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql
index 64b89ec0e..c78b946ed 100644
--- a/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql
+++ b/src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql
@@ -26,6 +26,7 @@ RETURNS boolean AS $$
     RETURN EXISTS (
       SELECT 1 FROM citus_internal_global_blocked_processes()
         WHERE waiting_global_pid = mBlockedGlobalPid
+        AND blocking_global_pid != waiting_global_pid
     ) OR EXISTS (
       -- Check on the workers if any logical replication job spawned by the
       -- current PID is blocked, by checking it's application name
```

```diff
 step s2-coordinator-create-index-concurrently:
  CREATE INDEX CONCURRENTLY select_table_index ON select_table(id);
-
+ <waiting ...>
 step s1-commit-worker:
  SELECT run_commands_on_session_level_connection_to_node('COMMIT');

 run_commands_on_session_level_connection_to_node
 ------------------------------------------------

 (1 row)

+step s2-coordinator-create-index-concurrently: <... completed>
```

The total test time was similar to what it was before this patch: 210 seconds for 100 consecutive runs of `isolation_select_vs_all_on_mx`

Reply:
but, as I noted earlier, we probably shouldn't do this change in `citus_isolation_test_session_is_blocked`. We should re-create the `citus_isolation_test_session_is_blocked ` function in a step just before the permutations which includes `step s2-coordinator-create-index-concurrently`, and have a comment on why we do this weird trick

Reply:
>  tried and tested another change after your comment, and it did not work. 1 test failed out of 200 runs.

That looks suspicious. It might indicate another bug. The weird thing is that the test you show failed run concurrently with `SELECT`, which should not be blocked anyway. And, could be really related to this one: https://github.com/citusdata/citus/pull/3877#discussion_r439335259

I wonder if thats just a random local process that `CREATE INDEX CONCURRENTLY` needs to wait (and shows up as blocking itself)

We can add one more filter ` AND blocking_global_pid != waiting_global_pid AND blocking_global_pid != 0`

But, as I noted earlier, we cannot merge this change to `citus_isolation_test_session_is_blocked ` for all, but only need to re-create this function when needed with these extra filters.   Or, could even introduce `citus_isolation_test_session_is_blocked_skip_self_local_blocks` UDF, where we override `citus_isolation_test_session_is_blocked` with `citus_isolation_test_session_is_blocked_skip_self_local_blocks` when needed

Reply:
I opened a PR #5939 and tried to implement the suggested solution above. I still observe some flakiness. Please take a look at the PR description that also includes a sample failing test output.